src/foam/dao/EasyDAO.js

```diff
@@ -162,13 +203,43 @@ if ( getContextualize() ) {
     build();
 }
 
+if ( getOrder() != null &&
+     getOrder().length > 0 ) {
+  // TODO: CompositeDAO or thenBy
+  for ( foam.mlang.order.Comparator comp : getOrder() ) {
+    delegate = delegate.orderBy(comp);
+  }
+}
+
+if ( getAuthorize() ) {
+  if ( foam.nanos.auth.Authorizable.class.isAssignableFrom(getOf().getObjClass()) ) {
+    delegate = new foam.nanos.auth.AuthorizationDAO(getX(), delegate);
+    setAuthenticate(false);
+  } else {
+    logger.warning("EasyDAO", "authorize=true but 'of' ",getOf().getId(), "not Authorizable");
+  }
+}
+
 if ( getAuthenticate() ) {
   delegate = new foam.dao.AuthenticatedDAO(
-    getName(),
+    getPermissionPrefix(),
     getAuthenticateRead(),
     delegate);
 }
 
+if ( getNSpec() != null &&
+     getNSpec().getServe() &&
+     ! getAuthorize() &&
+     ! getAuthenticate() &&
+     ! getReadOnly() ) {
```

Comment:
why is ReadOnly important? Isn't it a common case when served service is not readOnly? Like all the daos the client is writing to?

Reply:
For 'serve'd daos, they should be authenticate, authorized, or read-only.   Nothing should be open read-write.